modules/steam/steam.nix

```diff
@@ -146,6 +146,34 @@ let
 
     set -e
 
+    # OOBE handling.
+    # (See steam-jupiter.sh in steam-jupiter-oobe and steam-jupiter-stable)
+    # On first boot, we need to start whatever steam we have packaged, to
+    # allow the user to connect to the internet. We don't have the luxury
+    # that Steam's OOBE has to be heavy-handed. So instead we'll see if it
+    # looks like the user has never started steam. If it looks that way,
+    # we'll init OOBE, and otherwise undo it.
+    (
+      STEAM_LINKS="$HOME"/.steam
+      STEAM_DIR="$HOME"/.local/share/Steam
+      REGISTRY="$STEAM_LINKS"/registry.vdf
+
+      echo ":: Checking if we're in OOBE or not..."
+      if ! test -f "$REGISTRY"; then
```

Comment:
Maybe we can actually replace the check with this:

```
nmcli networking connectivity check
```

This may actually ensure an offline device will be able to start steam, even if steam was flagged as forced check, or whatever it was.

Reply:
What status would it show when a bootstrapped and previously connected Steam Deck is going to be turned on outside of stored Wi-Fi connection's coverage?

Reply:
not "full"; that status is always related to the *current instantaneous* reading of the check, *especially* when the `check` argument is given, it re-verifies.

So this would catch someone stepping out of home and ensure no update is tried to be ran.

We may also redundantly do the check for the steam directory being present, just as a preventative measure, since vendor does basically that.